app/api/v2/endpoints/rank_router.py

```
@@ -9,5 +9,13 @@ async def get_ranked_portfolios(req: RankDTOV2.GetRankPtfoRequest):
     try:
         rank_service = RankServiceV2()
         return rank_service.get_ranked_portfolios(req)
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)